rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -88,62 +85,96 @@ public class MavenPomDownloader {
     private boolean addLocalRepository;
 
     /**
-     * @param projectPoms    Other POMs in this project.
+     * @param ctx The execution context, which potentially contain Maven settings customization and {@link HttpSender} customization.
+     */
+    public MavenPomDownloader(ExecutionContext ctx) {
+        this(HttpSenderExecutionContextView.view(ctx).getHttpSender(), ctx);
+    }
+
+    /**
+     * A MavenPomDownloader for non-maven contexts where there are no project poms or assumption that maven central
+     * is implicitly added as a repository. In a Maven contexts, the regular constructor should be used instead
+     *
+     * @param ctx The execution context
+     */
+    public static MavenPomDownloader forNonMavenContext(ExecutionContext ctx) {
```

Comment:
I am wondering if we should maybe unconditionally set `addCentralRepository` and `addLocalRepository` to `false`.

I also think it would be nice to have two (or three?) factory methods instead of constructors. But this can be done in a separate step.

Reply:
I think it makes sense to do it in scope of this task as we deprecate all the `projectPom` parameters in constructors and remove them in calls to factory methods